src/ctapipe/tools/process.py

```diff
@@ -249,8 +294,24 @@ def should_compute_muon_parameters(self):
         if self.write.write_muon_parameters:
             return True
 
+        if (
+            self.calibrate.process_flatfield_events
+            or self.calibrate.process_pedestal_events
+        ):
+            return False
+
         return False
 
+    def should_process(self, event):
```

Comment:
There is already an event type filter in the tool that basically does the same and should be used for this.

Reply:
Good point!